.circleci/config.yml

```diff
@@ -1,69 +1,62 @@
-version: 2
-jobs:
-  test-python3:
+version: 2.1
+
+executors:
+  tox:
     docker:
-      - image: circleci/python:3.6.7
-      
+      - image: themattrix/tox
     working_directory: ~/repo
 
+commands:
+  tox:
+    parameters:
+      env:
+        type: string
     steps:
       - checkout
+      - run: tox -e << parameters.env >>
 
-      - restore_cache:
-          keys:
-          - v1-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "test-requirements-mypy.txt" }}
-          - v1-dependencies-
-
-      - run:
-          name: Run Tests
-          command: |
-            sudo pip install tox
-            tox -e mypy
-
-      - save_cache:
-          paths:
-            - ./.tox/py36
-          key: v1-dependencies-{{ checksum "requirements.txt" }}-{{ checksum "test-requirements-mypy.txt" }}
-
-      - run:
-          name: Upload Coverage
-          command: |
-            pip install --user codecov
-            python -m codecov
+jobs:
 
-  test-python2:
-    docker:
-      - image: circleci/python:2.7.15
+  lint:
```

Comment:
Is it possible to make this more concise by just invoking tox which by default executes all environments? Or is the idea to leverage CircleCI parallelism (are jobs parallelized)?

Reply:
Yes, all the jobs are parallelized. Also, if we used a single tox invocation we'd only get a single "check" on our GH PRs.

Reply:
Cool - very nice.